sklearn/utils/graph.py

```diff
@@ -14,11 +14,19 @@
 from scipy import sparse
 
 from ..metrics.pairwise import pairwise_distances
+from _param_validation import Interval, Integral, validate_params
 
 
 ###############################################################################
 # Path and connected component analysis.
 # Code adapted from networkx
+@validate_params(
+    {
+        "graph": ["array-like", "sparse matrix"],
+        "source": [Interval(Integral, 0, None, closed="left")],
+        "cutoff": [Interval(Integral, 1, None, closed="left"), None],
```

Comment:
We want to allow paths of length 0 because a node can be adjacent to itself
```suggestion
        "cutoff": [Interval(Integral, 0, None, closed="left"), None],
```